storage/storage.go

```diff
@@ -43,13 +43,45 @@ type TupleIterator interface {
 type Writes = []*openfgapb.TupleKey
 type Deletes = []*openfgapb.TupleKey
 
+// QueryOptions represent constraints that can be applied to a query to modify the response
+// behavior of the query.
+type QueryOptions struct {
```

Comment:
How many options do you envision having? If we only will have `limit` perhaps the extra complexity is not worth it?

Reply:
I envision having all of the options we use in our different Read, ReadUserTuple, and ReadUsersetTuples, etc.. I think we mostly use LIMIT and GROUP BY, but we may use a couple others. These are modifiers/constraints for the query in addition to the filters.